src/Graphex.hs

```diff
@@ -71,3 +73,11 @@ restrictTo (Graph m) keep = Graph . flip Map.mapMaybeWithKey m $ \k' v ->
   (if Set.notMember k' keep then Nothing else nonNullSet (Set.intersection keep v))
     where
         nonNullSet v = if Set.null v then Nothing else Just v
+
+-- | Convert a Graph to a Tree. If there is a cycle, treat the cycle point as a leaf.
+graphToTree :: Text -> Graph -> Tree Text
+graphToTree startingAt (Graph rels) = go mempty startingAt
+  where
+    go seen node =
+      let children = Set.toList $ Map.findWithDefault mempty node rels
+      in Tree.Node node $ if (Set.member node seen) then [] else fmap (go (Set.insert node seen)) children
```

Comment:
Do we need this cycle code?  My normal generators won't hit this, but I guess it wouldn't be hard to just add a link from the end to the beginning and verify that treeing it produces the right set of nodes.

Reply:
Yeah in practice it won't come up (especially analyzing Haskell codebases).

I guess I was just being defensive and handling any possible `Graph` here? It'll even handle `Graph $ Map.singleton "loop" "loop"`. Felt wrong to have a `_|_` lurking.